lib/ukcpio/cpio.c

```diff
@@ -70,7 +101,13 @@ extract_file(const char *path, const char *contents, size_t len,
 
 	uk_pr_info("Extracting %s (%zu bytes)\n", path, len);
 
-	fd = uk_syscall_r_open(path, O_CREAT | O_WRONLY | O_TRUNC, 0);
+	fd = uk_syscall_r_open(path, O_CREAT | O_WRONLY | O_EXCL, 0);
+	if (fd == -EEXIST) {
+		if ((err = try_remove(path)))
```

Comment:
I am not a fan of this style of programming. I would separate this assignment of `err` outside the `if` and then check with an `unlikely`... Also give a reason (through a comment) saying why a `uk_pr_warn` is enough and it's not a stop reason.

Reply:
Seeing that this isn't covered by our coding conventions, nor is it reported by checkpatch, I think it is very much a matter of taste, and I am a fan of this style when the assignments are short and concise.

Reply:
Ok but the separation would be worth here so that we could add `unlikely` I would say :). I would not add the function call inside the `unlikely`.

Reply:
You'd be surprised :stuck_out_tongue: `if (unlikely((r = try_remove(path))))`
But sure, if it helps readability that much.